lib/db.js

```diff
@@ -118,7 +117,6 @@ const legalOptionNames = [
  * @param {number} [options.bufferMaxEntries=-1] Sets a cap on how many operations the driver will buffer up before giving up on getting a working connection, default is -1 which is unlimited.
  * @param {(ReadPreference|string)} [options.readPreference] The preferred read preference (ReadPreference.PRIMARY, ReadPreference.PRIMARY_PREFERRED, ReadPreference.SECONDARY, ReadPreference.SECONDARY_PREFERRED, ReadPreference.NEAREST).
  * @param {object} [options.pkFactory] A primary key factory object for generation of custom _id keys.
- * @param {object} [options.promiseLibrary] A Promise library class the application wishes to use such as Bluebird, must be ES6 compatible
```

Comment:
This is technically a breaking change. I think we'll have to relegate this whole change to `master`, or think of a way where you can have promise providers everywhere that `promiseLibrary` is accepted in options.

Reply:
We could add in setters to the global promise lib when this option is passed in, however I believe It would still be a breaking change because AFAIK you could have a changeStream promise library and a gridFs promise library in the same runtime. Using a global promise library means it's changed for every method, not scoped to it's specific library. Thoughts?

Reply:
I think the ideal approach for users would be to deprecate the `promiseLibrary` option in 3.6, remove it in master for everything except `MongoClient` (keeping that deprecated), and then remove it from MongoClient in the next major release.